File: Python/List.py
from Mahasiswa import Mahasiswa

class List:
    __listMahasiswa = []

    def tambahdataMhs(self, MhsBaru):
        self.__listMahasiswa.append(MhsBaru)

    def tampilkanDataMhs(self):
        if (len(self.__listMahasiswa) == 0):
            print("Daftar Mahasiswa Kosong.")
        else:
            print("=============== DAFTAR MAHASISWA ==============\n\n")
            it = 0
            while(it < len(self.__listMahasiswa)):
                print("-------------------- " + str((it + 1)) + " --------------------");
                print("NIK             : " + self.__listMahasiswa[it].getNIK());
                print("NIM             : " + self.__listMahasiswa[it].getNIM());
                print("Nama            : " + self.__listMahasiswa[it].getNama());
                print("Jenis Kelamin   : " + self.__listMahasiswa[it].getJenisKelamin());
                print("Program Studi   : " + self.__listMahasiswa[it].getProdi());
                print("Fakultas        : " + self.__listMahasiswa[it].getFakultas());
                print("Asal Universitas: " + self.__listMahasiswa[it].getAsalUniv());
                print("Email           : " + self.__listMahasiswa[it].getEmailUniv());
                print("-------------------- " + str((it + 1)) + " --------------------\n");
                it = it + 1;

File: CPP/List.cpp
#include <bits/stdc++.h>
#include "Mahasiswa.cpp"

using namespace std;

class List
{
private:
    list<Mahasiswa> listMahasiswa;

public:
    void tambahDataMhs(Mahasiswa mhsbaru)
    {
        listMahasiswa.push_back(mhsbaru);
    }

    void tampilkanDataMhs()
    {
        if (listMahasiswa.empty())
        {
            cout << "Daftar Mahasiswa Kosong." << endl;
        }
        else
        {
            cout << "=============== DAFTAR MAHASISWA ==============\n"
                 << endl;

            int counter = 0;
            for (list<Mahasiswa>::iterator it = listMahasiswa.begin(); it != listMahasiswa.end(); it++)
            {
                cout << ("-------------------- ") << (counter + 1) << (" --------------------") << "\n";
                cout << ("NIK             : ") << (it->getNIK()) << "\n";
                cout << ("NIM             : ") << (it->getNIM()) << "\n";
                cout << ("Nama            : ") << (it->getNama()) << "\n";
                cout << ("Jenis Kelamin   : ") << (it->getJenisKelamin()) << "\n";
                cout << ("Program Studi   : ") << (it->getProdi()) << "\n";
                cout << ("Fakultas        : ") << (it->getFakultas()) << "\n";
                cout << ("Asal Universitas: ") << (it->getAsalUniv()) << "\n";
                cout << ("Email           : ") << (it->getEmailUniv()) << "\n";
                cout << ("-------------------- ") << (counter + 1) << (" --------------------\n");
                counter++;
                cout << "\n";
            }
        }
    }
};